airflow/ui/src/components/TriggerDag/TriggerDAGForm.tsx

```diff
@@ -0,0 +1,205 @@
+                render={({ field }) => (
+                  <Input
+                    {...field}
+                    placeholder="yyyy-mm-ddThh:mm"
+                    size="sm"
+                    type="datetime-local"
+                  />
+                )}
+              />
+            </Box>
+
+            <Box>
+              <Text fontSize="md" mb={2}>
+                Run ID
+              </Text>
+              <Controller
+                control={control}
+                name="runId"
+                render={({ field }) => (
+                  <Input
+                    {...field}
+                    placeholder="Run id, optional - will be generated if not provided"
+                    size="sm"
+                  />
+                )}
+              />
+            </Box>
+
+            <Box mb={9}>
+              <Text fontSize="md" mb={2}>
+                Configuration JSON
+              </Text>
+              <Controller
+                control={control}
+                name="configJson"
+                render={({ field }) => (
+                  <Box>
+                    <CodeMirror
+                      {...field}
+                      basicSetup
+                      extensions={[json(), autocompletion(), lineNumbers()]}
+                      height="200px"
+                      onChange={(value) => {
+                        field.onChange(value); // Update the form state
+                        try {
+                          // Attempt to parse the value as JSON
+                          JSON.parse(value);
+                          setJsonError(undefined); // Clear error if JSON is valid
+                        } catch {
+                          setJsonError("Invalid JSON format."); // Set error message if invalid
+                        }
+                      }}
+                      style={{
+                        border: "1px solid #CBD5E0",
+                        borderRadius: "8px",
+                        outline: "none",
+                        padding: "2px",
+                      }}
+                      theme={
+                        colorMode === "dark"
+                          ? (githubDark as Extension)
+                          : (githubLight as Extension)
+                      }
+                    />
+                    {jsonError! ? (
+                      <Box color="red.500" mt={2}>
+                        <Text fontSize="sm">{jsonError}</Text>
+                      </Box>
+                    ) : undefined}
+                  </Box>
+                )}
+              />
+            </Box>
+          </VStack>
+        ) : undefined}
+      </VStack>
+
+      <Box as="footer" display="flex" justifyContent="flex-end">
+        <HStack w="full">
+          {hasFormChanged() && <Button onClick={() => reset()}>Reset</Button>}
+          <Spacer />
+          <Button
+            disabled={Boolean(jsonError)} // Disable if there's an error
```

Comment:
Let's just leave this disabled for now. Since we don't have the API endpoint yet. Let's also put a TODO comment here. Or even in a text box in the UI to avoid confusion from otehr developers.

Reply:
I think it is reasonable to disable trigger if form validation fails - maybe also because of other validations.

But fr the moment - as long as the endpoint to trigger to is missing, I'd think would be cool to have it working as a mock... in the other PR I did earlier I raised an alaert where you could see the parameter "how it would have been triggered" if the PAI would be there :-D

Reply:
Yeah, a console log of the future request body works for me